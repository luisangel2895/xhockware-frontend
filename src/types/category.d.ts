export interface Category {
  name: string;
  status: boolean;
}

export interface AllCategories {
  all: {
    name: string;
    status: boolean;
  };
  health: {
    name: string;
    status: boolean;
  };
  economy: {
    name: string;
    status: boolean;
  };
  tecnology: {
    name: string;
    status: boolean;
  };
  politics: {
    name: string;
    status: boolean;
  };
  ecology: {
    name: string;
    status: boolean;
  };
  social: {
    name: string;
    status: boolean;
  };
}
