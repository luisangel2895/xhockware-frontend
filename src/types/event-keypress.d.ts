export interface EventKeyPress {
  isTrusted: boolean;
  altKey: boolean;
  bubbles: boolean;
  cancelBubble: boolean;
  cancelable: boolean;
  charCode: number;
  code: string;
}
