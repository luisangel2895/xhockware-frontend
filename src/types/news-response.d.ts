export interface NewsResponse {
  status: string;
  totalResults: number;
  articles: News[];
}

export interface News {
  source: Source;
  author: string;
  category: string;
  title: string;
  description: string;
  url: string;
  urlToImage: string;
  publishedAt: string;
  content: string;
}

export interface Source {
  id: string;
  name: string;
}
