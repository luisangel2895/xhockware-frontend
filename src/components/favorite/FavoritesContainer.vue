<template lang="pug">
.favorites-container(v-if="getFavoriteNews.length")
	favorite(v-for="(favorite, index) in getFavoriteNews" :key="favorite.source.id" :news="favorite")
.default-message(v-if="!getFavoriteNews.length") You dont have any favorite news.
</template>
<script lang="ts">
import { Options, Vue } from "vue-class-component";
import { mapGetters } from "vuex";
import Favorite from "@/components/favorite/Favorite.vue";

@Options({
  components: {
    Favorite,
  },
  computed: mapGetters(["getFavoriteNews"]),
})
export default class FavoritesContainer extends Vue {}
</script>

<style lang="scss" scoped>
.favorites-container {
  width: 100%;
  display: flex;
  flex-wrap: wrap;
  margin-top: 30px;
  margin-bottom: 40px;
}
.default-message {
  margin-top: 20px;
  width: 100%;
  text-align: center;
  font-family: $regular;
  font-size: 1.4rem;
  color: $grey_primary;
}

// PC
@media (min-width: 1300px) {
  .favorites-container {
    height: 300px;
    overflow-y: auto;
    &::-webkit-scrollbar {
      width: 3px;
      background-color: $grey_primary;
    }
    &::-webkit-scrollbar-track {
      box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.3);
      border-radius: 10px;
      background-color: #f5f5f5;
    }

    &::-webkit-scrollbar-thumb {
      border-radius: 10px;
      box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.3);
      background-color: $grey_primary;
    }
  }
}
</style>
