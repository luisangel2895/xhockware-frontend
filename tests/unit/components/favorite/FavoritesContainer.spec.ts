import { shallowMount } from "@vue/test-utils";
import { createStore } from "vuex";
// Components
import FavoritesContainer from "@/components/favorite/FavoritesContainer.vue";
// Mocks
import { favoritesNewsMock } from "../../mocks/news.mocks";
// Types
import { News } from "@/types/news-response";
interface State {
  favoriteNews: News[];
}

const initialStore = createStore({
  state(): State {
    return {
      favoriteNews: [] as News[],
    };
  },
  getters: {
    getFavoriteNews: (state) => (state as State).favoriteNews,
  },
});

const initialWrapper = shallowMount(FavoritesContainer, {
  global: {
    plugins: [initialStore],
  },
});

const store = createStore({
  state(): State {
    return {
      favoriteNews: favoritesNewsMock as News[],
    };
  },
  getters: {
    getFavoriteNews: (state) => (state as State).favoriteNews,
  },
});

const wrapper = shallowMount(FavoritesContainer, {
  global: {
    plugins: [store],
  },
});

describe("Test in view FavoritesContainer.vue", () => {
  // Start state of this component
  it("Match with snapshot initial", () => {
    expect(initialWrapper.html()).toMatchSnapshot();
  });
  it("Show default message when the app start or when we dont have any favorite news", () => {
    expect(initialWrapper.find(".default-message").exists()).toBeTruthy();
    expect(initialWrapper.find(".default-message").text()).toBe(
      "You dont have any favorite news."
    );
  });
  // When we have a favorites news
  it("Match with snapshot running", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("Checking the number of component (favorite) created by the received (getFavoriteNews)", () => {
    expect(wrapper.findAll("favorite-stub").length).toBe(
      favoritesNewsMock.length
    );
  });
});
