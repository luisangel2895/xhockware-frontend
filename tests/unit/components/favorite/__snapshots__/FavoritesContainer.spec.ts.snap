// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in view FavoritesContainer.vue Match with snapshot initial 1`] = `
<!--v-if-->
<div class="default-message">You dont have any favorite news.</div>
`;

exports[`Test in view FavoritesContainer.vue Match with snapshot running 1`] = `
<div class="favorites-container">
  <favorite-stub news="[object Object]"></favorite-stub>
  <favorite-stub news="[object Object]"></favorite-stub>
</div>
<!--v-if-->
`;
